tests/StreamTest.php

```diff
@@ -110,55 +112,58 @@ public function testProvidesStreamPosition()
         $stream->close();
     }
 
-    public function testCanDetachStream()
+    public function testDetachStreamAndClearProperties()
     {
-        $r = fopen('php://temp', 'w+');
-        $stream = new Stream($r);
-        $stream->write('foo');
-        $this->assertTrue($stream->isReadable());
-        $this->assertSame($r, $stream->detach());
-        $stream->detach();
-
-        $this->assertFalse($stream->isReadable());
-        $this->assertFalse($stream->isWritable());
-        $this->assertFalse($stream->isSeekable());
+        $handle = fopen('php://temp', 'r');
+        $stream = new Stream($handle);
+        $this->assertSame($handle, $stream->detach());
+        $this->assertTrue(is_resource($handle), 'Stream is not closed');
+        $this->assertNull($stream->detach());
 
-        $throws = function (callable $fn) use ($stream) {
-            try {
-                $fn($stream);
-                $this->fail();
```

Comment:
is test written wrongly as this also throws an exception which is caught. so it never failed when it should: `$stream->getSize()` returns null instead of throwing an exception when detached